_stbt/imgutils.py

```diff
@@ -157,6 +158,97 @@ def _frame_repr(frame):
         return "<%dx%d>" % (frame.shape[1], frame.shape[0])
 
 
+class Color:
+    """A BGR color, optionally with an alpha (transparency) value.
+
+    A Color can be created from an HTML-style hex string:
+
+    >>> Color('#f77f00')
+    Color('#f77f00')
+
+    Or from Blue, Green, Red values in the range 0-255:
+
+    >>> Color(0, 127, 247)
+    Color('#f77f00')
+
+    Note: When you specify the colors in this way, the BGR order is the
+    opposite of the HTML-style RGB order. This is for backwards compatibility
+    with the way OpenCV stores colors.
+    """
+    def __init__(self, *args):
+        self.array = None
+        if len(args) == 1:
+            if isinstance(args[0], Color):
+                self.array = args[0].array
+            elif isinstance(args[0], str):
+                self.array = Color._from_string(args[0])
+            elif isinstance(args[0], (list, tuple)) and len(args[0]) in (3, 4):
+                self.array = Color._from_sequence(*args[0])
+        elif len(args) in (3, 4):
+            self.array = Color._from_sequence(*args)  # pylint:disable=no-value-for-parameter
+        if self.array is None:
+            raise TypeError("Color: __init__() expected a Color, '#rrggbb' "
+                            "string, or 3 integers in Blue-Green-Red order. ")
+        self.hexstring = (
+            "#{0:02x}{1:02x}{2:02x}{3}".format(
```

Comment:
It might be better to make this a method - someone could update `self.array`, but this would still show the old value.

```suggestion

        def hexstring(self):
            return "#{0:02x}{1:02x}{2:02x}{3}".format(
```

Reply:
I think of these as immutable, but I don't know how to enforce that. Really all the members are private but I didn't want to go adding underscores everywhere. I suppose I'll just do it. I think of this as a private class that we need to expose for the sake of documentation / Sphinx — in practice users don't ever need to construct a `Color` explicitly.